src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -678,13 +713,23 @@ private void readInvalidLineFile(BinaryData invalidLineFileBinary, InvalidLineFi
                 if (jsonPath.toFile().exists()) {
                     logResponseFileContents(jsonPath);
                     ArmResponseInvalidLineRecord armResponseInvalidLineRecord = objectMapper.readValue(jsonPath.toFile(), ArmResponseInvalidLineRecord.class);
-                    UploadNewFileRecord uploadNewFileRecord = readInputJson(armResponseInvalidLineRecord.getInput());
+                    String input = armResponseInvalidLineRecord.getInput();
+                    UploadNewFileRecord uploadNewFileRecord = readInputJson(input);
                     if (nonNull(uploadNewFileRecord)) {
-                        armBatchResponses.addResponseBatchData(Integer.valueOf(uploadNewFileRecord.getRelationId()),
-                                                               armResponseInvalidLineRecord, invalidLineFileFilenameProcessor);
+                        if (StringUtils.isNotEmpty(uploadNewFileRecord.getRelationId())) {
+                            armBatchResponses.addResponseBatchData(Integer.valueOf(uploadNewFileRecord.getRelationId()),
+                                                                   armResponseInvalidLineRecord, invalidLineFileFilenameProcessor);
+
+                        } else {
+                            log.warn("Unable to get EOD id (relation id) from uploadNewFileRecord {} invalid line file {}",
+                                     armResponseInvalidLineRecord.getInput(), invalidLineFileFilenameProcessor.getInvalidLineFilename());
+                        }
+                    } else {
+                        log.warn("Failed to obtain EOD id (relation id) from invalid line record {} from file {}", input,
+                                 invalidLineFileFilenameProcessor.getInvalidLineFilename());
```

Comment:
This code seems to be repeated a fair few times wonder if this could be pulled out into its own method?

Reply:
The pattern is similar but the objects retrieved are different and each type has there own methods. Its simple enough to just follow the logic.

Reply:
Its more from a maintainability and testability view point having the logic centralised means updates are centralised and changes do not get missed

Reply:
Its unnecessary testing.

Reply:
Sorry I mean having centralised logic instead of duplicate code means we only need to test it once instead of in every place it is used so we gain much high quality tests.

and by having centralised logic in production code it reduces the risk of defects and makes changes easier as you only need to change / test one place instead of all instances which means it becomes easier to miss stuff

Reply:
@Ben-Edwards-cgi do you mean to merge with other methods e.g. `readUploadFile` in the same class which looks very similar to this one? If so, I tend to agree that would be a good idea. This class is nearly a 1000 lines long now which is way too many. Anything to make things cleaner is probably good from a maintainability point, especially for another developer picking this up in future.

Reply:
Yeah exactly that just so the logic is in a method we can re-use to prevent us from duplicating

Reply:
This is a bug ticket. To optimise here is going to take time and 3 other tickets are awaiting this fix. If this is so necessary raise another to optimise but this needs to go in ASAP.

Reply:
Dont think code optimisation/refactoring should be done part of this ticket. There should be a separate ticket to refactor the class and shouldnt be done part of a bug ticket which is blocking 6 other tickets.

Reply:
Yeh happy for a ticket to be raised to track this and get this one merged

Reply:
Created ticket to refactor: https://tools.hmcts.net/jira/browse/DMP-4414